src/main/kotlin/org/rust/ide/intentions/import/ImportNameIntention.kt

```diff
@@ -38,8 +36,16 @@ class ImportNameIntention : RsElementBaseIntentionAction<ImportNameIntention.Con
         // TODO: support reexports
         val candidates = RsNamedElementIndex.findElementsByName(project, basePath.referenceName)
             .filterIsInstance<RsQualifiedNamedElement>()
+            .filter { basePath != path || !(it is RsMod || it is RsModDeclItem || it.parent is RsMembers) }
             .mapNotNull {
                 val info = (it as? RsVisible)?.canBeImported(pathSuperMods) ?: return@mapNotNull null
+                // check that result after import can be resolved
+                // if no, don't add it in candidate list
+                if (basePath != path) {
+                    val pkg = info.target.pkg
+                    val fullPath = "${pkg.name}::${info.crateRelativePath.removeSuffix(basePath.text)}${path.text}"
+                    resolveStringPath(fullPath, pkg.workspace, project) ?: return@mapNotNull null
```

Comment:
Hm, an excellent idea! I think we need to go a bit further though and to try to actually insert a fake use declaration and resolve it. This is what `resolveStringPath` does under the hood actually, but it always inserts the path into the root module, and we preferably should use the current module instead.